From 69b9441eab2a7215509687dc22b48b6f212d22aa Mon Sep 17 00:00:00 2001
From: Rob Bradford <rob@linux.intel.com>
Date: Thu, 4 Jun 2009 15:43:20 +0100
Subject: [PATCH] =?utf-8?q?bgo#584832=20=E2=80=93=20Duplicate=20the=20exec=20string=20returned=20by=20gtk=5Frecent=5Finfo=5Fget=5Fapplication=5Finfo?=
MIME-Version: 1.0
Content-Type: text/plain; charset=utf-8
Content-Transfer-Encoding: 8bit

This function states that the caller is responsible for freeing the string
passed returned by reference. Unfortunately if you do this you get a crash
since the internal value is returned without being duplicated.
---
 gtk/gtkrecentmanager.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

Upstream-Status: Pending

diff --git a/gtk/gtkrecentmanager.c b/gtk/gtkrecentmanager.c
index 317b3d5..d062572 100644
--- a/gtk/gtkrecentmanager.c
+++ b/gtk/gtkrecentmanager.c
@@ -1764,7 +1764,7 @@ gtk_recent_info_get_application_info (GtkRecentInfo  *info,
     }
   
   if (app_exec)
-    *app_exec = ai->exec;
+    *app_exec = g_strdup (ai->exec);
   
   if (count)
     *count = ai->count;
-- 
1.6.3.1

